core/modules/css/CssAccess.cc

```diff
@@ -0,0 +1,848 @@
+/*
+ * LSST Data Management System
+ * Copyright 2015 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "css/CssAccess.h"
+
+// System headers
+#include <fstream>
+#include <sstream>
+
+// Third-party headers
+#include "boost/property_tree/ptree.hpp"
+#include "boost/property_tree/json_parser.hpp"
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+// Qserv headers
+#include "css/constants.h"
+#include "css/CssError.h"
+#include "css/EmptyChunks.h"
+#include "css/KvInterface.h"
+#include "css/KvInterfaceImplMem.h"
+#include "util/IterableFormatter.h"
+
+namespace {
+
+LOG_LOGGER _log = LOG_GET("lsst.qserv.css.CssAccess");
+
+}
+
+namespace lsst {
+namespace qserv {
+namespace css {
+
+std::shared_ptr<CssAccess>
+CssAccess::makeMemCss(std::string const& mapPath, std::string const& emptyChunkPath) {
+    std::ifstream f(mapPath);
+    if(f.fail()) {
+        throw ConnError();
+    }
+    return CssAccess::makeMemCss(f, emptyChunkPath);
+}
+
+std::shared_ptr<CssAccess>
+CssAccess::makeMemCss(std::istream& mapStream, std::string const& emptyChunkPath) {
+    return CssAccess::makeKvCss(std::make_shared<KvInterfaceImplMem>(mapStream),
+                                emptyChunkPath);
+}
+
+std::shared_ptr<CssAccess>
+CssAccess::makeKvCss(std::shared_ptr<KvInterface> const& kv, std::string const& emptyChunkPath) {
+    return std::shared_ptr<CssAccess>(new CssAccess(kv, std::make_shared<EmptyChunks>(emptyChunkPath)));
+}
+
+
+// Construct from KvInterface instance and empty chunk list instance
+CssAccess::CssAccess(std::shared_ptr<KvInterface> const& kvInterface,
+                     std::shared_ptr<EmptyChunks> const& emptyChunks,
+                     std::string const& prefix)
+    : _kvI(kvInterface), _emptyChunks(emptyChunks), _prefix(prefix) {
+}
+
+int
+CssAccess::cssVersion() {
+    return VERSION;
+}
+
+std::vector<std::string>
+CssAccess::getDbNames() const {
+    std::string p = _prefix + "/DBS";
+    return _kvI->getChildren(p);
+}
+
+bool
+CssAccess::containsDb(std::string const& dbName) const {
+    if (dbName.empty()) {
+        LOGF(_log, LOG_LVL_DEBUG, "Empty database name passed.");
```

Comment:
I agree, by default (and default usually means INFO) at this level there should not be anything printed at all. Higher level code may selectively decide to print something with info but low-level packages should be silent.